From a372c0e2673b3cacb43f8159d8badf6c9bb395c8 Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Fri, 31 Jul 2015 02:20:58 -0700
Subject: [PATCH] PlatformWindow: Add needed support in PlatformWindow.

This patch adds necessary API support in PlatformWindow. The changes
need to be evaluated further before trying to upstream them.
---
 ui/platform_window/platform_window.h | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/ui/platform_window/platform_window.h b/ui/platform_window/platform_window.h
index 6e6ace1..9042316 100644
--- a/ui/platform_window/platform_window.h
+++ b/ui/platform_window/platform_window.h
@@ -6,12 +6,16 @@
 #define UI_PLATFORM_WINDOW_PLATFORM_WINDOW_H_
 
 #include "base/memory/scoped_ptr.h"
+#include "base/strings/string16.h"
+#include "third_party/skia/include/core/SkPath.h"
 #include "ui/base/cursor/cursor.h"
 
 namespace gfx {
 class Rect;
 }
 
+class SkPath;
+
 namespace ui {
 
 class PlatformWindowDelegate;
@@ -22,8 +26,24 @@ class PlatformWindowDelegate;
 // underlying platform windowing system (i.e. X11/Win/OSX).
 class PlatformWindow {
  public:
+  enum PlatformWindowType {
+    PLATFORM_WINDOW_UNKNOWN,
+    PLATFORM_WINDOW_TYPE_TOOLTIP,
+    PLATFORM_WINDOW_TYPE_POPUP,
+    PLATFORM_WINDOW_TYPE_MENU,
+    PLATFORM_WINDOW_TYPE_BUBBLE,
+    PLATFORM_WINDOW_TYPE_WINDOW,
+    PLATFORM_WINDOW_TYPE_WINDOW_FRAMELESS
+  };
+
   virtual ~PlatformWindow() {}
 
+  virtual void InitPlatformWindow(PlatformWindowType type,
+                                  gfx::AcceleratedWidget parent_window) { }
+  virtual void SetWidgetTitle(const base::string16& title) { }
+  virtual void SetWindowShape(const SkPath& path) { }
+  virtual void SetOpacity(unsigned char opacity) { }
+
   virtual void Show() = 0;
   virtual void Hide() = 0;
   virtual void Close() = 0;
-- 
1.9.1

